### PR TITLE
added Graveyard, deck size, isSilenced

### PR DESCRIPTION
**Pull request by AiSatan**
I've added  Graveyard, deck size and isSilenced

Also I fixed null exception in mulligan

### CODE REVIEW

File: src/Hearthrock.Contracts/RockMinion.cs [hunk 1]
```diff
@@ -1,132 +1,138 @@
...
-        /// </summary>
-        public bool HasLifesteal { get; set; }
-
-        /// <summary>
-        ///  Gets or sets a value indicating whether the minion is poisonous.
-        /// </summary>
-        public bool IsPoisonous { get; set; }
-    }
-}
+﻿// <copyright file="RockMinion.cs" company="https://github.com/yangyuan">
```
Comment (yangyuan) [inline src/Hearthrock.Contracts/RockMinion.cs]: It's very weird that git says entire this file is changed.
Probably caused by complex commits relations.

Do you mind to rebase and squash the changes into 1 commit? (Which is a required practice in many big companies)

File: src/Hearthrock/Engine/RockEngineAction.cs [hunk 2]
```diff
@@ -138,13 +138,16 @@ public void ApplyAll()
         {
             foreach (int cardId in this.actions)
             {
-                var card = this.pegasus.GetObject(cardId);
-                if (card.CardId == "GAME_005")
+                var card = this.pegasus?.GetObject(cardId);
```
Comment (yangyuan) [inline src/Hearthrock/Engine/RockEngineAction.cs]: this.pegasus can be null? It doesn't make sense.
I prefer to undo the changes in here, if "this.pegasus" could be null, I should fix it.

BTW I prefer to fix null instead of tolerating null. Because I prefer to have fewer states. An uncontrollable number of states usually lead to bugs.

File: src/Hearthrock.Contracts/RockMinion.cs (hunk 1 above)
Comment (AiSatan) [reply src/Hearthrock.Contracts/RockMinion.cs]: oh, yes, sure, I will do it asap

File: src/Hearthrock/Engine/RockEngineAction.cs (hunk 2 above)
Comment (AiSatan) [reply src/Hearthrock/Engine/RockEngineAction.cs]: it's happens sometime, and mulligan do not works as expected, I do not why it happens, it's really strange. yes sure I will undo this, maybe we will find the real reason.
